packages/utilities/src/EnvFileUtils.ts

```diff
@@ -0,0 +1,107 @@
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { existsSync, readFileSync } from "fs";
+import { homedir } from "os";
+import { join } from "path";
+import { ImperativeError } from "../../error/src/ImperativeError";
+import { EnvironmentalVariableSettings } from "../../imperative/src/env/EnvironmentalVariableSettings";
+import * as JSONC from "comment-json";
+
+/**
+ * Utility to load environment JSON files and set variables
+ * @export
+ * @class EnvFileUtils
+ */
+export class EnvFileUtils {
+
+    /**
+     * Check and read in an environment file from the user home directory using the app name
+     * If the file is valid, set the environment variables
+     * If the file is not valid, display an error and continue
+     * @param {string} appName - The application name
+     * @param {boolean} checkCliHomeVariableFirst - Check inside of *_CLI_HOME first if it is defined
+     * @param {string} envPrefix - The environment variable prefix
+     * @returns {void}
+     * @throws {ImperativeError}
+     */
+    public static setEnvironmentForApp(appName: string, checkCliHomeVariableFirst: boolean = false, envPrefix?: string) {
+        const expectedFileLocation = this.getEnvironmentFilePath(appName, checkCliHomeVariableFirst, envPrefix);
+        if (expectedFileLocation) {
+            try {
+                const fileContents = readFileSync(expectedFileLocation).toString(); // Read the file in
+                const fileContentsJSON = JSONC.parse(fileContents);
+                Object.keys(fileContentsJSON).forEach( key => {
+                    process.env[key] = fileContentsJSON[key];
+                });
+            } catch (err) {
+                let errorMessage = "Failed to set up environment variables from the environment file.\n" +
+                    "Environment variables will not be available.\nFile: " + expectedFileLocation;
+
+                if (err.line != null && err.column != null) {
+                    errorMessage += "\nLine: " + err.line.toString() + "\nColumn: " + err.column.toString();
+                }
+
+                throw new ImperativeError({msg: errorMessage, causeErrors: err});
+            }
+        }
+    }
+
+    /**
+     * Get the expected path for the user's environment variable file
+     * @param {string} appName - The application name
+     * @param {boolean} checkCliHomeVariableFirst - Check inside of *_CLI_HOME first if it is defined
+     * @param {string} envPrefix - environment variable prefix
+     * @returns {string|undefined} - Returns the path if it exists, or undefined if it does not
+     */
+    public static getEnvironmentFilePath(appName: string, checkCliHomeVariableFirst: boolean = false, envPrefix?: string): string | undefined {
```

Comment:
Curious if we could save some typing-definition if these methods returned `null` instead `undefined`